From 75ba4a547282f91d653872a4bba5f5eae234ea6c Mon Sep 17 00:00:00 2001
From: rofl0r <retnyg@gmx.net>
Date: Wed, 22 Jan 2014 00:57:48 +0100
Subject: [PATCH 2/3] libc-compat.h: prevent redefinition of struct ethhdr

---

Note: Modified for support backport to linux_libc_headers_3.18.131, to
match the linux_quic_3.18.131. This is for building kernel 3.18.131
with Yocto 2.5, which has a preferred libc header of 4.15.x

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Modified-by: Steven MacDonald <smacdonald@sierrawireless.com>
Upstream-Status: Submitted

 include/uapi/linux/if_ether.h    | 4 +++-
 include/uapi/linux/libc-compat.h | 6 ++++++
 2 files changed, 9 insertions(+), 1 deletion(-)

Index: linux-4.15/include/uapi/linux/libc-compat.h
===================================================================
--- linux-4.15.orig/include/uapi/linux/libc-compat.h	2018-02-05 17:40:42.338370731 -0500
+++ linux-4.15/include/uapi/linux/libc-compat.h	2018-02-05 17:40:42.334370603 -0500
@@ -51,6 +51,12 @@
 
 #ifndef __KERNEL__ /* we're used from userspace */
 
+#ifdef _NETINET_IF_ETHER_H /* musl */
+#define __UAPI_DEF_ETHHDR 0
+#else /* glibc uses __NETINET_IF_ETHER_H, and includes the kernel header. */
+#define __UAPI_DEF_ETHHDR 1
+#endif
+
 /* Coordinate with libc net/if.h header. */
 #if defined(_NET_IF_H)
 
